plugins/user/joomla/joomla.php

```diff
@@ -335,16 +345,18 @@ public function onUserLogout($user, $options = [])
 		$my      = Factory::getUser();
 		$session = Factory::getSession();
 
+		$userid = (int) $user['id'];
+
 		// Make sure we're a valid user first
-		if ($user['id'] == 0 && !$my->get('tmp_user'))
+		if ($user['id'] === 0 && !$my->get('tmp_user'))
```

Comment:
```suggestion
		if ($userid === 0 && !$my->get('tmp_user'))
```
`$user['id']` is from database.